packages/compass-connections-navigation/src/connections-navigation-tree.tsx

```diff
@@ -378,68 +394,56 @@ const ConnectionsNavigationTree: React.FunctionComponent<
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-expect-error ignoring test props so they are not part of the interface
   __TEST_REACT_WINDOW_OVERSCAN = null,
-  ...restProps
 }) => {
-  // we'll have either connections for MC version, or databases for SC version
-  const {
-    isSingleConnection,
-    data,
-  }: {
-    isSingleConnection: boolean;
-    data:
-      | MCConnectionsNavigationTreeProps['connections']
-      | SCConnectionsNavigationTreeProps['databases'];
-  } = useMemo(() => {
-    if ((restProps as SCConnectionsNavigationTreeProps).databases) {
-      return {
-        isSingleConnection: true,
-        data: (restProps as SCConnectionsNavigationTreeProps).databases,
-      };
-    }
-    return {
-      isSingleConnection: false,
-      data: (restProps as MCConnectionsNavigationTreeProps).connections,
-    };
-  }, [restProps]);
+  const isSingleConnection = !usePreference(
+    'enableNewMultipleConnectionSystem'
+  );
 
   const listRef = useRef<List | null>(null);
   const id = useId();
 
   useEffect(() => {
     if (activeNamespace) {
-      onDatabaseExpand(activeNamespace, true);
+      for (const connection of connections) {
+        onDatabaseExpand(connection.connectionInfo.id, activeNamespace, true);
+      }
     }
-  }, [activeNamespace, onDatabaseExpand]);
+  }, [activeNamespace, onDatabaseExpand, connections]);
```

Comment:
this `connections` dependency is still causing issues in the legacy version. it is not possible to collapse active connection, because that changes the connections array and triggers this expand.
Let's restrict it to the first connection, this way the legacy version works as before, and we can fix the multi-connection version once we have active connection.
```suggestion
  }, [activeNamespace, onDatabaseExpand, connections[0].connectionInfo.id]);
```